### ISSUE TITLE
wlsunset SystemD service not starting

### ISSUE THREAD

**Issue by dmayle** [nix-community/home-manager#1865]
### Issue description

I'm a new nixos and home manager user (and relatively new to systemd as well, though I have a long history with linux from before systemd).

I have installed the wlsunset service using the module here, and I can't get it to start, either on system start or on login.  Manually starting works.

I also tried updating the Install value of the systemd service (either to graphical-session.target or multi-user.target) and that didn't work either.

#### Maintainer CC

@matrss 

#### Technical details

```
{ config, pkgs, ... }:
let
  unstableHomeManager = fetchTarball https://github.com/nix-community/home-manager/tarball/07f6c6481e0cbbcaf3447f43e964baf99465c8e1;
  unstable = builtins.fetchTarball https://github.com/nixos/nixpkgs/tarball/0b876eaed3ed4715ac566c59eb00004eca3114e8;
  unstablePkgs = import unstable { config.allowUnfree = true; };
in
{
  imports =
    [
      "${unstableHomeManager}/modules/services/wlsunset.nix"
    ];

  # Let Home Manager install and manage itself.
  programs.home-manager.enable = true;

  # Home Manager needs a bit of information about you and the
  # paths it should manage.
  home.username = "douglas";
  home.homeDirectory = "/home/douglas";

  services.wlsunset = {
    package = unstablePkgs.wlsunset;
    enable = true;
    # Nice, France
    latitude = "43.7";
    longitude = "7.2";
  };
  systemd.user.services.wlsunset.Install = { WantedBy = [ "multi-user.target" ]; };

  # This value determines the Home Manager release that your
  # configuration is compatible with. This helps avoid breakage
  # when a new Home Manager release introduces backwards
  # incompatible changes.
  #
  # You can update Home Manager without changing this value. See
  # the Home Manager release notes for a list of state version
  # changes in each release.
  home.stateVersion = "20.09";
}
```

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.104, NixOS, 20.09.3454.60b18a066e8 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixos-20.09.3454.60b18a066e8"`
 - channels(douglas): `"home-manager-20.09"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by berbiche** [comment]
Hi and welcome to NixOS and Home Manager.

The systemd target `multi-user.target` is only a valid target for the system's services.
For user services, you should use `graphical-session.target` or `default.target`.

I assume you are using Sway or another wlroots-based compositor if you are using wlsunset.
You would need to create a target that starts binds to `graphical-session.target`.
If you use Sway you can enable automatically starting systemd services with an `[Install] WantedBy = graphical-session.target` with the option `wayland.windowManager.sway.systemdIntegration = true;`.

<br>
<br>
<br>

If you are not using Sway, you can take inspiration from these:
https://github.com/nix-community/home-manager/blob/07f6c6481e0cbbcaf3447f43e964baf99465c8e1/modules/services/window-managers/i3-sway/sway.nix#L289
https://github.com/nix-community/home-manager/blob/07f6c6481e0cbbcaf3447f43e964baf99465c8e1/modules/services/window-managers/i3-sway/sway.nix#L406-L414
Systemd says to never start the `graphical-session.target`, one must use a target bound to `graphical-session.target` like in the example above.

**Comment by dmayle** [comment]
@berbiche thank you so much!

Last night, as I was falling asleep, I started to wonder if the problem was with my sway config, but without your comment I never would have figured it out.

I actually have sway configured in /etc/nixos/configuration.nix, so no access to the home-manager module.

I already had the sway session created (following the wiki, I think), I just needed to add that line to my sway.conf:
```
  environment.etc."sway/config".text = (builtins.readFile "${pkgs.sway}/etc/sway/config") + ''
      exec "systemctl --user import-environment; systemctl --user start sway-session.target"
    '';
```
